src/Medidata.ZipkinTracer.Core/TraceProvider.cs

```diff
@@ -72,11 +73,11 @@ internal TraceProvider(IZipkinConfig config, IOwinContext context = null)
                 headerSampled = context.Request.Headers[SampledHeaderName];
             }
 
-            TraceId = Parse(headerTraceId) ? headerTraceId : GenerateHexEncodedInt64FromNewGuid();
-            SpanId = Parse(headerSpanId) ? headerSpanId : TraceId;
-            ParentSpanId = Parse(headerParentSpanId) ? headerParentSpanId : string.Empty;
+            TraceId = headerTraceId.IsParsableTo128Or64Bit() ? headerTraceId : GenerateHexFromNewGuid(config.Create128BitTraceId);
```

Comment:
so there's two things going on:

1. GenerateHexFromNewGuid should be tested particularly that the exact formatting is known (as opposed to presumed). This was the point I was making about ensuring it doesn't have hyphens etc.

2. It is pretty common to either restart a trace when headers passed to you malformed, or silently generate a new one. I've seen it done both ways and there are valid arguments some of them captured here: https://github.com/openzipkin/b3-propagation/issues/3#issuecomment-248189315